webpack/assets/javascripts/react_app/routes/ExternalRoutes.js

```diff
@@ -0,0 +1,25 @@
+import { Route } from 'react-router-dom';
+import React from 'react';
+import { addGlobalFill } from '../components/common/Fill/GlobalFill';
+import { renderRoute } from './RoutingService';
+
+/**
+ * Adds a plugin's routes into core
+ * @param  {String} id  plugin's id - can be its name
+ * @param  {Array}   routes an array that contains a plugin's routes
+ */
+export const registerRoutes = (id, routes) =>
+  routes.map(({ render, path, beforeRendering = null, ...routeProps }, index) =>
```

Comment:
getting the error: `Uncaught TypeError: beforeRenderingCallback is not a function`
can you remove the `null` here?

```suggestion
  routes.map(({ render, path, beforeRendering, ...routeProps }, index) =>
```